internal/config/identity/openid/jwt.go

```diff
@@ -48,19 +47,25 @@ func (pk *publicKeys) parseAndAdd(b io.Reader) error {
 		return err
 	}
 
-	pk.Lock()
-	defer pk.Unlock()
-
 	for _, key := range jwk.Keys {
-		pk.pkMap[key.Kid], err = key.DecodePublicKey()
+		pkey, err := key.DecodePublicKey()
 		if err != nil {
 			return err
 		}
+		pk.add(key.Kid, pkey)
```

Comment:
`pk` is already locked here, so the `pk.Lock`  in the add method can never be taken.

Reply:
it isnb't @klauspost not sure where do you see it.. `pk.add()` is the only write lock held..